rfc-drafts/stream.md

```diff
@@ -346,19 +506,42 @@ where
 ```
 
 This is a "conversion" trait such that anything which implements `Stream` can also implement 
-`Attached Stream`.
+`Lending Stream`.
 
 This trait captures the case we re-use internal buffers. This would be less flexible for 
-consumers, but potentially more efficient. Types could implement the `AttachedStream` 
+consumers, but potentially more efficient. Types could implement the `LendingStream` 
 where they need to re-use an internal buffer and `Stream` if they do not. There is room for both.
 
 We would also need to pursue the same design for iterators - whether through adding two traits
 or one new trait with a "conversion" from the old trait.
 
 This also brings up the question of whether we should allow conversion in the opposite way - if
-every "Detached" stream can become an attached one, should _some_ detached streams be able to 
-become attached ones? These use cases need more thought, which is part of the reason 
-it is out of the scope of this particular RFC.
+every non-lending stream can become a lending one, should _some_ non-lending streams be able to 
+become lending ones? 
+
+We can say that, as the Rust language stands today, we cannot cleanly convert impl Stream to impl LendingStream due to a coherence conflict.
+
+If you have other impls like:
+
+```rust
+impl<T> Stream for Box<T> where T: Stream
+```
+
+and
+
+```rust
+impl<T> LendingStream for Box<T> where T: LendingStream
+```
+
+There is a coherence conflict for `Box<impl Stream>`, so presumably it will fail the coherence rules. 
+
+[More examples are available here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a667a7560f8dc97ab82a780e27dfc9eb).
```

Comment:
This is a really useful reference; with the context of the text in this RFC I feel like this is the best explainer I've encountered so far.

Reply:
Thanks to @nikomatsakis!